### PR TITLE
test: increase `pydantic\aliases.py` coverage to 100%

### PR DESCRIPTION
**Pull request by AlessandroMiola**
## Change Summary

Add a tiny case to increase coverage of `pydantic\aliases.py` to 100%.
Tests for forbidden string indexing of `AliasPath` argument via the `search_dict_for_path` utility. 

note: I thought that the newly added assertions could fit into the existing `test_validation_alias_parse_data`, therefore I didn't add a new test function. Happy to change if you think otherwise (or get guidance in any case)!

## Related issue number

Contributes to https://github.com/pydantic/pydantic/issues/7656.

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Review by sydney-runkle — CHANGES_REQUESTED** [review]
Hi @AlessandroMiola,

Thanks for your contribution here!

Could we extract this logic into a separate test, maybe called `test_search_dict_for_path` or something similar? I think that'll help to keep things organized.

Awesome work on increasing coverage here!

**Comment by sydney-runkle** [comment]
As a side note, I wonder if we can add a more clear `coverage` integration to PRs that adds a comment like the `codspeed` integration to celebrate increases in test coverage + warn against regressions :). I suppose the CI job is fine for now, but if anyone is interested, this could be a cool CI focused task.

**Comment by AlessandroMiola** [comment]
> As a side note, I wonder if we can add a more clear `coverage` integration to PRs that adds a comment like the `codspeed` integration to celebrate increases in test coverage + warn against regressions :). I suppose the CI job is fine for now, but if anyone is interested, this could be a cool CI focused task.

I'd would be nice, indeed. I'll try to dig into it :)

File: tests/test_aliases.py [hunk 1]
```diff
@@ -585,6 +586,14 @@ class Model(BaseModel):
     assert Model.model_fields['x'].validation_alias == value
 
 
+def test_search_dict_for_alias_path():
+    class Model(BaseModel):
+        x: str = Field(validation_alias=AliasPath('a', 1))
+
+    assert Model.model_fields['x'].validation_alias.search_dict_for_path({'a': ['hello', 'world']}) == 'world'
+    assert Model.model_fields['x'].validation_alias.search_dict_for_path({'a': 'hello'}) is PydanticUndefined
```
Comment (sydney-runkle) [inline tests/test_aliases.py:594]:
```suggestion
    class Model(BaseModel):
        ap = AliasPath('a', 1)
        assert ap.search_dict_for_path({'a': ['hello', 'world']}) == 'world'
        assert ap.search_dict_for_path({'a': 'hello'}) is PydanticUndefined
```

Little simplification, we don't need the model abstraction here :0

File: tests/test_aliases.py [hunk 2]
```diff
@@ -585,6 +586,13 @@ class Model(BaseModel):
     assert Model.model_fields['x'].validation_alias == value
 
 
+def test_search_dict_for_alias_path():
+    class Model(BaseModel):
+        ap = AliasPath('a', 1)
+        assert ap.search_dict_for_path({'a': ['hello', 'world']}) == 'world'
+        assert ap.search_dict_for_path({'a': 'hello'}) is PydanticUndefined
```
Comment (sydney-runkle) [inline tests/test_aliases.py:593]:
```suggestion
    ap = AliasPath('a', 1)
    assert ap.search_dict_for_path({'a': ['hello', 'world']}) == 'world'
    assert ap.search_dict_for_path({'a': 'hello'}) is PydanticUndefined
```

Comment (sydney-runkle) [reply tests/test_aliases.py:593]: Oops, correction

**Review by sydney-runkle — APPROVED** [review]

**Pull request merged**